java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4973,6 +4973,218 @@ public void xgroupCreate_xgroupDestroy(BaseClient client) {
+        String streamId1_1 = "1-1";
+        String streamId1_2 = "1-2";
+        String streamId1_3 = "1-3";
+        LinkedHashMap<String, String> streamMap = new LinkedHashMap();
+        streamMap.put("f1", "v1");
+        streamMap.put("f2", "v2");
+
+        assertEquals(
+                streamId1_0,
+                client.xadd(key, streamMap, StreamAddOptions.builder().id(streamId1_0).build()).get());
+        assertEquals(
+                streamId1_1,
+                client
+                        .xadd(key, Map.of("f3", "v3"), StreamAddOptions.builder().id(streamId1_1).build())
+                        .get());
+        assertEquals(
+                streamId1_2,
+                client
+                        .xadd(key, Map.of("f4", "v4"), StreamAddOptions.builder().id(streamId1_2).build())
+                        .get());
+        assertEquals(OK, client.xgroupCreate(key, groupName1, streamId0_0).get());
+
+        Map result =
+                client
+                        .xreadgroup(
+                                Map.of(key, ">"),
+                                groupName1,
+                                consumer1,
+                                StreamReadGroupOptions.builder().count(1L).build())
+                        .get();
+        assertDeepEquals(
+                Map.of(key, Map.of(streamId1_0, new String[][] {{"f1", "v1"}, {"f2", "v2"}})), result);
+
+        // Sleep to ensure the idle time value and inactive time value returned by xinfo_consumers is >0
+        Thread.sleep(2000);
+        Map[] consumers = client.xinfoConsumers(key, groupName1).get();
+        assertEquals(1, consumers.length);
+        Map consumerInfo = consumers[0];
+        assertEquals(consumer1, consumerInfo.get("name"));
+        assertEquals(1L, consumerInfo.get("pending"));
+        assertTrue((Long) consumerInfo.get("idle") > 0L);
+
+        if (REDIS_VERSION.isGreaterThanOrEqualTo("7.2.0")) {
+            assertTrue((Long) consumerInfo.get("inactive") > 0L);
+        }
+
+        // Test with GlideString
+        consumers = client.xinfoConsumers(gs(key), gs(groupName1)).get();
+        assertEquals(1, consumers.length);
+        consumerInfo = consumers[0];
+        assertEquals(consumer1, consumerInfo.get("name"));
+        assertEquals(1L, consumerInfo.get("pending"));
+        assertTrue((Long) consumerInfo.get("idle") > 0L);
+
+        if (REDIS_VERSION.isGreaterThanOrEqualTo("7.2.0")) {
+            assertTrue((Long) consumerInfo.get("inactive") > 0L);
+        }
+
+        // Create consumer2 and read the rest of the entries with it
+        assertTrue(client.xgroupCreateConsumer(key, groupName1, consumer2).get());
+        result = client.xreadgroup(Map.of(key, ">"), groupName1, consumer2).get();
+        assertDeepEquals(
+                Map.of(
+                        key,
+                        Map.of(
+                                streamId1_1, new String[][] {{"f3", "v3"}},
```

Comment:
test same with glidestring too

Reply:
Do we need to test GlideString for readgroup here? I'm testing GlideStrings for info groups and consumers already.